enclave_mgmt/create_models.py

```diff
@@ -411,3 +450,27 @@ def quiz_attempts_and_multichoice_responses_model(
             .to_dict(orient='records'):
         QuizAttemptMultichoiceResponses.model_validate(item)
     return quiz_attempts_df, quiz_attempt_multichoice_responses_df
+
+
+def filter_dataframes_by_course_id(research_filter_df, unfiltered_df):
+    filtered_df = pd.merge(
+        research_filter_df, unfiltered_df, on='course_id'
+    )
+    return filtered_df
+
+
+def drop_duplicate_users(enrollments_df, users_df):
```

Comment:
I think this method name is misleading. It is actually something like `filter_users_by_enrollments` or something like that. As part of the implementation of that filtering, it ensures duplicates aren't introduced.

Reply:
I've updated the name to `filter_users_by_enrollments`.